cmd/log.go

```diff
@@ -0,0 +1,62 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/redhat-developer/odo/pkg/application"
+	"github.com/redhat-developer/odo/pkg/component"
+	"github.com/redhat-developer/odo/pkg/project"
+	"github.com/spf13/cobra"
+)
+
+var logCmd = &cobra.Command{
+	Use:   "log [component_name]",
+	Short: "Retrieve the log for the given component.",
+	Long:  `Retrieve the log for the given component.`,
+	Example: `  # Get the logs for the nodejs component,
+  odo log nodejs
+	`,
+	Args: cobra.RangeArgs(0, 1),
+	Run: func(cmd *cobra.Command, args []string) {
+
+		// Retrieve stdout / io.Writer
+		stdout := os.Stdout
+
+		// Retrieve the client
+		client := getOcClient()
+
+		// Application
+		currentApplication, err := application.GetCurrent(client)
+		checkError(err, "")
+
+		// Project
+		currentProject := project.GetCurrent(client)
+
+		// Retrieve and set the current component if none passed
+		var currentComponent string
+		if len(args) == 0 {
+
+			// Get the current compoment
+			currentComponent = getComponent(client, "", currentApplication, currentProject)
+
+		} else {
+
+			currentComponent = args[0]
```

Comment:
we don't need this whole `else` section. This is already taken care of in `getComponent` function